### ISSUE TITLE
Remove global monkeypatch of Tensor/Module repr; use safe logging

### ISSUE THREAD

**Issue by jashvira** [jashvira/policy-gradient-experiments#2]
Problem
- Replaces torch.Tensor.__repr__/__str__ and nn.Module.__repr__ globally to suppress large prints.
- Brittle, may break third-party libs, harms debugging performance.

Code
```python
# experiments/grpo/grpo_train.py
...
    torch.Tensor.__repr__ = clean_tensor_repr
    torch.Tensor.__str__ = clean_tensor_str
    torch.nn.Module.__repr__ = clean_module_repr
...
```

Impact
- Unexpected behaviour for libs expecting canonical reprs.
- Potential slowdown; harder debugging when details are needed.

Proposed fix
- Remove monkeypatching entirely.
- Use torch.set_printoptions(threshold=..., edgeitems=...) and targeted logging helpers.

Acceptance criteria
- No monkeypatch of Torch core types.
- Only safe print/logging options used.
